Sources/protoc-gen-swiftgrpc/Generator-Client.swift

```diff
@@ -164,6 +168,12 @@ extension Generator {
           println("/// Asynchronous. Unary.")
           println("func \(methodFunctionName)(_ request: \(methodInputName), completion: @escaping (\(methodOutputName)?, CallResult) -> Void) throws -> \(callName)")
         }
+        if rxSwiftCode {
```

Comment:
Consider calling through to the protocol's official asynchronous unary implementation (see line 169); that way all this code could be put into a protocol extension and would not need to be part of the main protocol. That would also have the advantage of letting you provide a default argument for `metadata` (provided the `metadata` argument gets added to the unary call's method signature first):

```
func \(methodFunctionName)(_ request: \(methodInputName), metadata customMetadata: Metadata?) -> Observable<\(methodOutputName)> {
  // use metadata: customMetadata ?? self.metadata when calling through
}
```